netty/src/main/java/io/grpc/netty/InternalNettyChannelBuilder.java

```diff
@@ -57,14 +54,23 @@ public static void overrideAuthorityChecker(
   }
 
   /**
-   * Creates a custom client transport that allows overriding the protocol negotiator.
+   * Interface to create netty dynamic parameters.
    */
-  public static ConnectionClientTransport newClientTransport(
-      ClientTransportFactory transportFactory, SocketAddress serverAddress, String authority,
-      String userAgent, ProtocolNegotiator negotiator) {
-    // Casting to avoid making {@link NettyTransportFactory} public.
-    return ((NettyTransportFactory) transportFactory)
-        .newClientTransport(serverAddress, authority, userAgent, negotiator);
+  public interface DynamicParamsFactory extends NettyChannelBuilder.DynamicParamsFactory {
+    @Override
+    DynamicParams create(SocketAddress targetServerAddress, String authority, String userAgent);
+  }
+
+  /**
+   * {@link DynamicParams} are those that may depend on late-known information about a client
+   * transport.  This interface can be used to dynamically alter params based on the params of
+   * {@code ClientTransportFactory}.
```

Comment:
Since this is specifically for overriding the parameters of `ClientTransportFactory.newClientTransport()`, I would make it more obvious through naming. `DynamicParams` is a bit vague. How about `TransportCreationParamsFilter`. It would be better to also have a reference to `newClientTransport()` in the javadoc.